### PR TITLE
Definition of what we aim to accomplish with our code

### PR DESCRIPTION
**Pull request by SpenserJ**
The bot would like to request that we all try to meet a few conditions to keep the code cleaner. I think you'll like how its worded. It isn't a requirement that any code meets this, but we should aspire to write clean, testable (and tested) code.

If you want to downvote this, please explain why.

### CODE REVIEW

**Comment by botwillacceptanything** [comment]
#### :ballot_box_with_check: Voting procedure reminder:
To cast a vote, post a comment containing `:+1:` (:+1:), or `:-1:` (:-1:).
Remember, you **must :star:star this repo for your vote to count.**

All comments within this discussion are searched for votes, regardless of the time of posting.
You can cast as many votes as you want, but only the last one will be counted.
(You may consider editing your comment instead of adding a new one.)
Comments containing both up- and down-votes are disregarded.
PR authors automatically count as a :+1: vote.

A decision will be made after this PR has been open for **15** minutes (plus/minus **10** percent, to avoid people timing their votes), and at least **8** votes have been made.
A supermajority of **65%** is required for the vote to pass.

*NOTE: the PR will be closed if any new commits are added after:* 3538554d1eca4e9e49e0495cc291b7099daa02b9

**Comment by botwillacceptanything** [comment]
:+1:

**Comment by dos1** [comment]
>If you want to downvote this, please explain why.

Makes me want to downvote it without explaining why.
Well, if I downvoted, it would already be an explanation, so I guess upvote is the only thing that's left now... 

Well played, sir. :+1:

**Comment by samis** [comment]
:+1:  Just remember to fix the 'three simple steps' bit afterwards.

**Comment by SpenserJ** [comment]
@samis Damn, I forgot about that! Good catch.
@dos1 Haha alright, well I may not understand the reasoning against it, but I appreciate the upvote anyway.

**Comment by gpersons** [comment]
:+1:

**Comment by RyanMilb** [comment]
:+1:

File: DoD.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# "Perfect World" Definition of Done
+
+Before I consider your PR to be finished, I ask that it meets these basic requirements. 
+
+1. My unit tests are passing.
+  * Run `gulp mocha`
```
Comment (shalecraig) [inline DoD.md]: If you re-roll this commit, please add a comment about `npm install`

File: DoD.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+1. You have thoroughly tested everything that you touched.
+1. My [code coverage](https://coveralls.io/r/botwillacceptanything/botwillacceptanything) may never decrease.
+  * If you added anything to my code, you'd better have a matching test!
+1. You've let everyone else know what you [changed](https://github.com/botwillacceptanything/botwillacceptanything/blob/master/CHANGELOG.md).
+1. Any [ideas](https://github.com/botwillacceptanything/botwillacceptanything/blob/master/ideas.md) that you just implemented are checked off.
+  * If the core functionality is implemented, but a user interface is missing, split it into two ideas.
+  * If the functionality paves the way for a new idea, add it to the list.
+
+How many of those did you check off?
```
Comment (shalecraig) [inline DoD.md]: If you re-roll this commit, please change this to be a sub-title:

`## How many of those did you check off?`

**Comment by shalecraig** [comment]
:+1: I think it's fine, if you end up re-submitting the PR, please fix my comments (but they're not blockers)

**Comment by thenightwassave** [comment]
:+1:

**Comment by SpenserJ** [comment]
Good call @shalecraig. Thanks for the input

**Comment by dbpokorny** [comment]
:+1:

**Comment by botwillacceptanything** [comment]
#### <a target="_blank" href="http://thecatapi.com/?id=1qd"></a>
:+1: The vote passed! This PR will now be merged into master.

----
**Tallies:**
:+1:: 8 (100%) 
:-1:: 0 (0%)

These users aren't stargazers, so their votes were not counted: 
 * @gpersons

**Comment by timweightman** [comment]
:+1:

**Pull request merged**